packages/aws-cdk-lib/aws-events-targets/lib/sns.ts

```diff
@@ -37,8 +52,19 @@ export class SnsTopic implements events.IRuleTarget {
    * @see https://docs.aws.amazon.com/eventbridge/latest/userguide/resource-based-policies-eventbridge.html#sns-permissions
    */
   public bind(_rule: events.IRule, _id?: string): events.RuleTargetConfig {
-    // deduplicated automatically
-    this.topic.grantPublish(new iam.ServicePrincipal('events.amazonaws.com'));
+    // role can't be passed when authorizeUsingRole is false
+    if (this.props.authorizeUsingRole === false && this.props.role) {
+      throw new ValidationError('Cannot provide a role when authorizeUsingRole is false', _rule);
+    }
+
+    // role is used when authorizeUsingRole is true or role is provided
+    const useRole = this.props.authorizeUsingRole || this.props.role;
+
+    // deduplicated automatically, only needed if role is not used
+    if (!useRole) this.topic.grantPublish(new iam.ServicePrincipal('events.amazonaws.com'));
+
+    const role = useRole ? this.props.role ?? singletonEventRole(_rule): undefined;
+    if (role) this.topic.grantPublish(role);
```

Comment:
This suggested approach avoids multiple checks for the same thing (`useRole`):

```suggestion
    role = undefined;

    if (this.props.authorizeUsingRole === true) {
        // role-based authorization
        const role = this.props.role ?? singletonEventRole(_rule);
        this.topic.grantPublish(role);
    } else {
       // role can't be passed when authorizeUsingRole is false
       if (this.props.role) {
             throw new ValidationError('Cannot provide a role when authorizeUsingRole is false', _rule);
       }
   
       this.topic.grantPublish(new iam.ServicePrincipal('events.amazonaws.com'));
    }
```